RELEASE_PROCEDURE.md

```diff
@@ -38,6 +38,8 @@ DLA-Future follows [Semantic Versioning](https://semver.org).
    with local repository. Exclude blocks delimited by `###` comments. These are only intended for the
    internal spack package.
 
+1. For patch releases: update `CHANGELOG` of `master` branch.
+
```

Comment:
I was thinking this could be "merge `version_X.Y` branch into master"`? It doesn't always work if we plan on ever patching old versions after we've released a new minor/major version, but will work in 99% of cases I would expect. You can maybe separate this change into a different PR to not block on this discussion?

Reply:
Yes, that is fair. https://github.com/eth-cscs/DLA-Future/pull/1250